### ISSUE TITLE
Failed to build docker image

### ISSUE THREAD

**Issue by Yegiri** [cdr/code-server#1919]
- Local OS: macOS 10.15.4

## Description
I tried to build my own docker image by running
```
./build.sh
```
When it runs
```
Step 8/14 : RUN ARCH="$(dpkg --print-architecture)" &&     curl -fsSL "https://github.com/boxboat/fixuid/releases/download/v0.4.1/fixuid-0.4.1-linux-$ARCH.tar.gz" | tar -C /usr/local/bin -xzf - &&     chown root:root /usr/local/bin/fixuid &&     chmod 4755 /usr/local/bin/fixuid &&     mkdir -p /etc/fixuid &&     printf "user: coder\ngroup: coder\n" > /etc/fixuid/config.yml
```
I always get this error:
```
curl: (18) transfer closed with 2254950 bytes remaining to read

gzip: stdin: unexpected end of file
tar: Unexpected EOF in archive
tar: Unexpected EOF in archive
tar: Error is not recoverable: exiting now
The command '/bin/sh -c ARCH="$(dpkg --print-architecture)" &&     curl -fsSL "https://github.com/boxboat/fixuid/releases/download/v0.4.1/fixuid-0.4.1-linux-$ARCH.tar.gz" | tar -C /usr/local/bin -xzf - &&     chown root:root /usr/local/bin/fixuid &&     chmod 4755 /usr/local/bin/fixuid &&     mkdir -p /etc/fixuid &&     printf "user: coder\ngroup: coder\n" > /etc/fixuid/config.yml' returned a non-zero code: 2
```

**Comment by code-asher** [comment]
You might have some kind of network issue if the transfer is getting aborted. Maybe setting a lower keepalive time on the `curl` command would help (`--keepalive-time 1` for example) but that's just a wild shot in the dark.

Are you able to curl the binary outside of Docker? Link: https://github.com/boxboat/fixuid/releases/download/v0.4.1/fixuid-0.4.1-linux-amd64.tar.gz

**Comment by Yegiri** [comment]
Thank you for your help！
I can't curl the binary outside of Docker, but I can download this file locally.
![image](https://user-images.githubusercontent.com/51883273/88696242-97e41300-d135-11ea-8c16-9bb6a51ee292.png)
and I still can't build the docker image.
![image](https://user-images.githubusercontent.com/51883273/88696727-2d7fa280-d136-11ea-85ce-415b4c1b2d15.png)
Maybe, it's my network issue?

**Comment by code-asher** [comment]
Yeah I think it's very likely a network issue. One thing you could try is to copy the tar.gz you downloaded instead of using `curl` in the Dockerfile (similar to how the `.deb` is copied a few lines down).

The file you downloaded looks odd though, it should be `fixuid-0.4.1-linux-amd64.tar.gz` but it ends in `12.42.11.gz`.

**Comment by Yegiri** [comment]
Which directory should I copy it to？

**Comment by Yegiri** [comment]
I solved the above problem.
But there is a new problem，how to generate this file?
```
Step 10/15 : COPY release-packages/code-server*.deb /tmp/
COPY failed: no source files were specified
```

**Comment by Yegiri** [comment]
My problem is solved, it is indeed a network issue, thank you very much!

**Comment by qinzehua** [comment]
> * Local OS: macOS 10.15.4
> 
> ## Description
> I tried to build my own docker image by running
> 
> ```
> ./build.sh
> ```
> 
> When it runs
> 
> ```
> Step 8/14 : RUN ARCH="$(dpkg --print-architecture)" &&     curl -fsSL "https://github.com/boxboat/fixuid/releases/download/v0.4.1/fixuid-0.4.1-linux-$ARCH.tar.gz" | tar -C /usr/local/bin -xzf - &&     chown root:root /usr/local/bin/fixuid &&     chmod 4755 /usr/local/bin/fixuid &&     mkdir -p /etc/fixuid &&     printf "user: coder\ngroup: coder\n" > /etc/fixuid/config.yml
> ```
> 
> I always get this error:
> 
> ```
> curl: (18) transfer closed with 2254950 bytes remaining to read
> 
> gzip: stdin: unexpected end of file
> tar: Unexpected EOF in archive
> tar: Unexpected EOF in archive
> tar: Error is not recoverable: exiting now
> The command '/bin/sh -c ARCH="$(dpkg --print-architecture)" &&     curl -fsSL "https://github.com/boxboat/fixuid/releases/download/v0.4.1/fixuid-0.4.1-linux-$ARCH.tar.gz" | tar -C /usr/local/bin -xzf - &&     chown root:root /usr/local/bin/fixuid &&     chmod 4755 /usr/local/bin/fixuid &&     mkdir -p /etc/fixuid &&     printf "user: coder\ngroup: coder\n" > /etc/fixuid/config.yml' returned a non-zero code: 2
> ```

hi, i aslo want to build a docker image locally, can you give me a example Dockfile?

**Comment by jsjoeio** [comment]
@qinzehua if you open a Discussion, we can help you!
